lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -13,6 +13,11 @@
     environment(HOME: '/root')
   end
 
+  has_command(:docker_compose, command(:dockercompose)) do
+    Dir.mkdir('/tmp_docker') unless Dir.exist?('/tmp_docker')
```

Comment:
I'm getting permission denied from this, even when puppet agent is running as root. Besides that, it seems wrong to be creating this directory in the root of the filesystem. Also, what is this for? The `TMPDIR` env var is never used in this code. Seems like this is here by mistake for the tests only.

Reply:
1. Unsure why you are getting permission denied, it works for me IRL and also passes acceptance tests.  What OS are you using?
2. /tmp_docker is in the base path so that it is adjacent to /tmp, which is the dir that docker-compose uses by default.  What location do you suggest?
3. the $TMPDIR env var is respected by the Python underlying docker-compose.  Using a path other than /tmp and setting that var overcomes the well-documented issue of docker-compose failing to run when noexec is set on /tmp.  The change is there to overcome this and not for testing purposes only.

Reply:
On Ubuntu. What is the issue that this pull request is trying to address?

Reply:
I wonder if the problem is that this code is running on the Puppet Server, not the agent, because the error occurs during catalog compilation. I can't find any evidence of this error other than during the puppet agent run, so I'm not sure why it's getting permission denied.

Reply:
Also, it would be good if this were somehow made optional, so that we don't end up with `/tmp_docker` on every system, even where docker isn't used.

Reply:
@kenyon to address your comments:
- the pull request addresses the known issue with docker-compose executing from /tmp as also reported at https://github.com/docker/compose/issues/8041
- agree that the issue, while appearing only on the first puppet run, may be trying to apply the dir on the server.  It may be linked to other reported issues as referenced on your issue raised as a result of this unexpected outcome
- the actual creation of the dir only occurs if resources are applied by docker_compose, and will not actually occur otherwise.  Further, if you have this module on your puppetserver but not applied to nodes, they do not appear to be affected.

I agree that it is both unexpected and need to be addressed, and have reported more detailed observations in your raised issue and also requested maintainer input.

Thank you for reporting the issue.